### PR TITLE
test(coverage): ratchet floor + add inventory & runtime contract tests

### PR DESCRIPTION
**Pull request by RC219805**
Item 1 (coverage floor): bump core-tier --cov-fail-under from 20% to 35%
and add scripts/ci/check_per_package_coverage.py to enforce per-package
floors on top: src/tp/ at 60%, lux_depth_v3/validators/ at 80%, the rest
of lux_depth_v3/ at 50%. The script reads the existing coverage.xml so
it adds no second test run. src/tp was previously absent from --cov
sources; add it so the floor is meaningful.

Item 2 (missing direct tests): add four offline, lazy-import-safe test
files for surfaces that were previously covered only transitively or
not at all — coreml_backend (allowlist + platform/dependency gating +
cache helpers), validators/run_card_integrity (helper-level / module
API / path-traversal hardening), DA3 runtime contract (venv path
constants + worker argv parser), Depth Pro runtime contract (license
metadata + checkpoint/python resolution precedence + worker argv).

Item 3 (route inventory): add tests/test_app_route_inventory.py that
walks app.routes, asserts every method+path pair is in an explicit
ROUTE_REGISTRY tagged by family, and enforces v1/v2 jobs API parity.
Drift (added or removed routes) becomes a visible diff that has to be
acknowledged in review.

https://claude.ai/code/session_01MWoDcETjeb7iRCaRZVfqyU

### CODE REVIEW

File: scripts/ci/check_per_package_coverage.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        covered = 0
+        valid = 0
+        normalized_prefix = prefix.replace("\\", "/")
+        for cls in classes:
+            filename = _normalize_filename(cls.attrib.get("filename", ""))
+            if not filename.startswith(normalized_prefix):
+                # Some coverage configurations strip the leading "src/" — try
+                # matching against the package path with that segment removed.
+                stripped = normalized_prefix.removeprefix("src/")
+                if not stripped or not filename.startswith(stripped):
```
[the remark this answers is not in the document]
Comment (RC219805) [reply scripts/ci/check_per_package_coverage.py]: Fixed in 61e9670. Replaced the bare `(prefix, floor)` tuples with a `PackageFloor(prefix, floor, exclude_prefixes=...)` dataclass and configured the `lux_depth_v3/` parent floor with `exclude_prefixes=("src/transformation_portal/lux_depth_v3/validators/",)`. Aggregate now skips files matched by any exclude prefix, so the parent rollup no longer double-counts validator files. Added a regression test `tests/test_check_per_package_coverage.py::TestAggregateNestedExclusion::test_parent_excludes_nested_floor` that constructs a synthetic XML where high validator coverage would inflate the parent percentage from 50% → 83% if double-counted; with the fix the parent correctly reports 50% and fails its 70% floor in the test. Sibling `test_parent_without_exclusion_double_counts` pins the inverse to guard against accidental short-circuiting.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -379,6 +379,18 @@ jobs:
...
+        # Uses Node 22's built-in --experimental-test-coverage. The
+        # --test-coverage-lines / --test-coverage-functions flags inside
+        # the script make the Node test runner exit non-zero if either
+        # threshold drops below the floor. Floors are intentionally
+        # conservative starters (50% line, 50% function); ratchet upward
+        # in package.json as coverage improves — never silently lower.
+        run: |
+          set -euo pipefail
+          cd web/secure-landing
+          npm run test:coverage
```
[the remark this answers is not in the document]
Comment (RC219805) [reply .github/workflows/build.yml]: Fixed in 61e9670. Replaced the two-step `npm test` + `npm run test:coverage` pair with a single "Run frontdoor Node tests with coverage floor" step that only runs `test:coverage`. Coverage runs the same tests, just with the `--test-coverage-lines=50 --test-coverage-functions=50` flags added — so we now build:portal once and run the suite once, while still enforcing the floors. Added a step comment explaining the rationale so the choice is visible to future editors.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/ci/check_per_package_coverage.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    ("src/tp/", 60.0),
+    # Run-card validators are the binding contract surface for governed deliverables.
+    ("src/transformation_portal/lux_depth_v3/validators/", 80.0),
+    # Lux Depth V3 core orchestrator seams (config_resolver, pipeline_coordinator,
+    # execution_engine, artifact_manager, manifest, provenance, run_card_contract,
+    # io_atomic, reconstruction_manifest). Backends and the very large
+    # segmentation_backend live alongside but are excluded by the deeper
+    # validators/ floor above and treated under the global floor.
+    ("src/transformation_portal/lux_depth_v3/", 50.0),
+)
```
[the remark this answers is not in the document]
Comment (RC219805) [reply scripts/ci/check_per_package_coverage.py:79]: Fixed in 61e9670 — went with the explicit `exclude_prefixes` list option you suggested. `PACKAGE_FLOORS` is now a tuple of `PackageFloor(prefix, floor, exclude_prefixes=...)` dataclasses; the `lux_depth_v3/` parent floor declares `exclude_prefixes=("src/transformation_portal/lux_depth_v3/validators/",)`. `aggregate()` skips files matched by any exclude prefix using the same `_matches_prefix` helper as the include side (so the `src/`-stripped fallback applies symmetrically). Module docstring updated to spell out the rationale, and `tests/test_check_per_package_coverage.py::TestAggregateNestedExclusion` pins both the with-exclusion and without-exclusion behaviours so the inflation can't silently come back.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/ci/check_per_package_coverage.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        if self.valid == 0:
+            return False
+        return self.percentage >= self.floor
+
+
+def _normalize_filename(raw: str) -> str:
+    """Cobertura filenames may be relative or absolute; normalize to forward slashes."""
+    return raw.replace("\\", "/").lstrip("./")
+
+
```
[the remark this answers is not in the document]
Comment (RC219805) [reply scripts/ci/check_per_package_coverage.py]: Fixed in 61e9670 — went with the "locate the last `/src/` segment and slice from there" option you suggested. `_normalize_filename` now:

1. converts backslashes to forward slashes,
2. strips a single leading `./` (only that exact prefix, not arbitrary dot/slash chars),
3. if the path is still absolute, finds the **last** occurrence of `/src/` and slices from there (using `rfind` so a repo whose path itself contains "src" — e.g. `/work/my-src-repo/src/tp/x.py` — still resolves to `src/tp/x.py` rather than picking up a fictional segment).

If an absolute path has no `/src/` segment at all, it's returned unchanged so the caller surfaces a clear "matched 0 covered source files" failure instead of silently truncating. Regression tests in `tests/test_check_per_package_coverage.py::TestNormalizeFilename` cover all four shapes (relative, `./`-prefixed, backslash, absolute), the src-in-repo-name edge case, and the absolute-without-src defensive path. `TestAggregateAbsolutePaths::test_absolute_filename_matches_relative_prefix` reproduces the original `/home/runner/work/.../src/tp/foo.py` scenario end-to-end through `aggregate()`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/ci/check_per_package_coverage.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            if lines is None:
+                continue
+            for line in lines.findall("line"):
+                valid += 1
+                hits = int(line.attrib.get("hits", "0"))
+                if hits > 0:
+                    covered += 1
+        results.append(PackageResult(prefix=prefix, floor=floor, covered=covered, valid=valid))
+    return results
+
```
[the remark this answers is not in the document]
Comment (RC219805) [reply scripts/ci/check_per_package_coverage.py]: Done in 61e9670 — added `tests/test_check_per_package_coverage.py` (20 tests, ~250 LOC) using a `_write_coverage` helper that builds tiny synthetic Cobertura XML inline. Coverage areas:

- `TestNormalizeFilename` (6 cases) — relative, `./`-prefixed, backslash, absolute, src-in-repo-name edge case, absolute-without-src-segment defensive path.
- `TestMatchesPrefix` (3 cases) — direct match, no-match, `src/`-stripped fallback.
- `TestAggregateBasic` (3 cases) — passing floor, failing floor, zero-match-is-failure (the typo-detection guarantee).
- `TestAggregateAbsolutePaths` (1 case) — end-to-end absolute-filename → relative-prefix matching through `aggregate()`.
- `TestAggregateNestedExclusion` (2 cases) — exclude_prefixes correctly subtracts nested files from a parent rollup, and a sibling test that pins the without-exclusion behaviour so accidentally short-circuiting `aggregate()` would fail.
- `TestMain` (3 cases) — exit code matrix: missing xml=2, failure=1, success=0.
- `TestRenderTable` (2 cases) — pass/fail row distinguishability and N/A rendering for zero-valid prefixes.

The test module loads the script via `importlib.util.spec_from_file_location` (registering in `sys.modules` so the in-script `@dataclass` can resolve its module — caught a `dataclasses._is_type` AttributeError on the first run). All 20 collect and pass under pytest 9 + Python 3.11.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/lux_depth_v3/test_coreml_backend.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+        # The allowlist is part of the CoreML release contract: adding a model
+        # here must be accompanied by a published .mlpackage and a runtime
+        # validation entry. Lock the membership so changes show up in review.
+        assert coreml_backend._SUPPORTED_PUBLISHED_COREML_MODEL_IDS == {
+            "apple/coreml-depth-anything-v2-small",
+        }
+
+    def test_unsupported_model_id_is_rejected_when_coremltools_missing(self, monkeypatch):
+        # Even with coremltools unavailable, the allowlist gate must fire
+        # before any conversion attempt.
```
[the remark this answers is not in the document]
Comment (RC219805) [reply tests/lux_depth_v3/test_coreml_backend.py]: Fixed in 61e9670. Renamed `test_unsupported_model_id_is_rejected_when_coremltools_missing` → `test_unsupported_model_id_is_rejected_before_load_or_convert` and rewrote the comment to spell out that the test asserts gate **ordering**: the allowlist gate fires before any `_load_or_convert` work begins. The body intentionally sets `COREML_AVAILABLE=True` so that the coremltools gate (which is its own separate test, `test_init_fails_closed_when_coremltools_unavailable`) doesn't short-circuit the path under exercise.

---
_Generated by [Claude Code](https://claude.ai/code)_
